### PR TITLE
Add local mobile support to iOS

### PR DESCRIPTION
**Pull request by mihaildu**
## Summary

This PR adds support for local mobile readers

## Motivation

We need this feature and it's already in stripe's roadmap. Full discussion here https://github.com/stripe/stripe-terminal-react-native/issues/439

## Testing

- [x] I tested this manually
- [ ] I added automated tests

To test I create a simple React Native app. I used `StripeTerminalProvider` and fetched a connection token from backend. Then just waited for Stripe Terminal to initialize, discovered all local mobile readers and connected to the first one. After connection I create a new payment intent, collect it and process it. I only got to the tap to pay screen, which I was not able to use because it asks me for a stripe test card to use in my test environment. So I guess there is still a small bit to test with processing and capturing the payment, but will only be able to do that later. My code looks something like this
```
// Screen.tsx
<StripeTerminalProvider
  tokenProvider={fetchTokenProvider}
>
  <App />
</StripeTerminalProvider>

// App.tsx
const App = () => {
  const {
    initialize,
    isInitialized,
    discoverReaders,
    discoveredReaders,
    connectLocalMobileReader,
    cancelDiscovering,
    createPaymentIntent,
    collectPaymentMethod,
    processPayment,
  } = useStripeTerminal();

  useEffect(() => {
    initialize();
  }, [initialize]);

  useEffect(() => {
    (async () => {
      if (isInitialized) {
        await discoverReaders({
          discoveryMethod: 'localMobile',
          simulated: false,
        });
      }
    })();
  }, [isInitialized, discoverReaders]);

  const testLocationId = '<test-location-id>';
  const testAcctId = '<acct-id>';

  const connectToLocalMobileReader = async reader => {
    const {reader: connectedReader} = await connectLocalMobileReader({
      reader,
      locationId: testLocationId,
      onBehalfOf: testAcctId,
    });
    
    const {paymentIntent} = await createPaymentIntent({
      amount: 100,
      currency: 'usd',
      onBehalfOf: testAcctId,
        transferDataDestination: testAcctId,
        applicationFeeAmount: 20,
        paymentMethodTypes: ['card_present'],
      });

    const {paymentIntent: collectedPaymentIntent} = await collectPaymentMethod({
      paymentIntentId: paymentIntent.id,
    });

    const {paymentIntent: processedPaymentIntent} = await processPayment(collectedPaymentIntent.id);
  }

  const discoveredReadersList = (
    <FlatList
      data={discoveredReaders.map(item => ({...item, key: item.label}))}
      renderItem={({item}: {item: Reader.Type}) => (
        <Text
          key={item.label}
          onPress={() => connectToLocalMobileReader(item)}
        >
          {item.label}
        </Text>
      )}
    />
  );

  return (
    <View>
      <Text>Readers on device (Local Mobile)</Text>
      {discoveredReadersList}
      <Button
        text="Cancel Discovering"
        onPress={async () => {
          await cancelDiscovering();
        }}
      />
    </View>
  );
};

```

## Documentation

Select one:

- [ ] I have added relevant documentation for my changes.
- [ ] This PR does not result in any developer-facing changes.

### CODE REVIEW

**Comment by billfinn-stripe** [comment]
Great, thanks! I pulled the branch, and we have to make a handful of additional changes to support local mobile support for Android and add support to the example and dev apps. Let's land this change into `main`, and then I can follow up with a PR that adds support to the dev and example apps.

Can you merge the latest from `main` into your branch to fix the CI build failure? If you have any trouble or questions, I can provide the git commands to run.

**Comment by mihaildu** [comment]
> Can you merge the latest from main into your branch to fix the CI build failure?

I rebased my branch on top of `main`. Bitrise seems stuck waiting for an action (I assume a PR approval). Let me know if there's anything else I have to do!

**Comment by mihaildu** [comment]
As a side note, there is also the [feat/add-capture-method-ios](https://github.com/mihaildu/stripe-terminal-react-native/tree/feat/add-capture-method-ios) branch that adds support for captureMethod (tested and it seems to work). I could open a PR for that as well!

**Review by billfinn-stripe — APPROVED** [review]

**Comment by ricardomalagon** [comment]
> 
Hi, I'm getting this status in the createPaymentIntent:
status: 'requiresPaymentMethod'
 
Any idea how to set the payment method as tap to pay? or how can I get to screen to process the payment.

**Pull request merged**
